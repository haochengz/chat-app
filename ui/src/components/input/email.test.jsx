
import React from 'react'
import EmailInput from './email'
import { shallow, mount } from 'enzyme'

describe('<EmailInput />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<EmailInput />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should emit value to the parent component', () => {
    let valueMock
    const inputMock = v => valueMock = v
    const wrapper = shallow(
      <EmailInput
        onChange={inputMock}
      />
    )
    const input = wrapper.find('InputItem')
    input.simulate('change', 'test')
    expect(valueMock).toEqual('test')
  })

  it('should emit errors if there is an invalid value in the input box', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <EmailInput
        onChange={change}
      />
    )
    const input = wrapper.find('InputItem')
    input.simulate('change', 'test')
    expect(errors).toHaveLength(1)
  })

  it('should clean the error once change the value to an valid value', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <EmailInput
        onChange={change}
      />
    )
    const input = wrapper.find('InputItem')
    input.simulate('change', 'iandme')
    expect(errors).toHaveLength(1)
    input.simulate('change', 'i@me.com')
    expect(errors).toHaveLength(0)
  })

  it('should had no error if the input box are empty and it was never changed yet', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    shallow(
      <EmailInput
        onChange={change}
      />
    )
    expect(errors).toEqual(null)
  })

  it('should emit an error if the input box was changed and then change back to empty', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <EmailInput
        onChange={change}
      />
    )
    const input = wrapper.find('InputItem')
    input.simulate('change', 'testuser@gmail.com')
    expect(errors).toHaveLength(0)
    input.simulate('change', '')
    expect(errors).toHaveLength(2)
  })

  it('should render an error badge if any errors passing in to the component', () => {
    const wrapper = mount(
      <EmailInput
        errors={['error1', 'error2']}
      />
    )
    expect(wrapper.exists('.am-input-error-extra')).toBe(true)
  })

  it('should no error badge if no errors passing in to the component', () => {
    const wrapper = mount(
      <EmailInput
        errors={[]}
      />
    )
    expect(wrapper.exists('.am-input-error-extra')).toBe(false)
  })

  it('should calls the showErrors method if error badge had been clicked and there are errors in error array', () => {
    const errorHandler = jest.fn()
    const wrapper = mount(
      <EmailInput
        errors={['error1', 'error2']}
      />
    )
    wrapper.instance().showErrors = errorHandler
    wrapper.instance().forceUpdate()
    const errorBadge = wrapper.find('.am-input-error-extra')
    errorBadge.simulate('click')
    expect(errorHandler).toBeCalled()
  })
})