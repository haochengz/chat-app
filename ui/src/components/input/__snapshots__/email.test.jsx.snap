// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailInput /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Email:",
      "clear": true,
      "disabled": false,
      "editable": true,
      "error": false,
      "extra": "",
      "labelNumber": 5,
      "moneyKeyboardAlign": "right",
      "moneyKeyboardWrapProps": Object {},
      "onBlur": [Function],
      "onChange": [Function],
      "onErrorClick": [Function],
      "onExtraClick": [Function],
      "onFocus": [Function],
      "onVirtualKeyboardConfirm": [Function],
      "placeholder": "Email",
      "prefixCls": "am-input",
      "prefixListCls": "am-list",
      "type": "text",
      "updatePlaceholder": false,
      "value": undefined,
    },
    "ref": null,
    "rendered": "Email:",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Email:",
        "clear": true,
        "disabled": false,
        "editable": true,
        "error": false,
        "extra": "",
        "labelNumber": 5,
        "moneyKeyboardAlign": "right",
        "moneyKeyboardWrapProps": Object {},
        "onBlur": [Function],
        "onChange": [Function],
        "onErrorClick": [Function],
        "onExtraClick": [Function],
        "onFocus": [Function],
        "onVirtualKeyboardConfirm": [Function],
        "placeholder": "Email",
        "prefixCls": "am-input",
        "prefixListCls": "am-list",
        "type": "text",
        "updatePlaceholder": false,
        "value": undefined,
      },
      "ref": null,
      "rendered": "Email:",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
